app/services/events.py

```diff
@@ -1,13 +1,31 @@
 import json
 import logging
 
+from eth_typing import ChecksumAddress, HexStr
+from hexbytes import HexBytes
 from safe_eth.eth.utils import fast_is_checksum_address
+from safe_eth.safe.multi_send import MultiSend
 
 from ..workers.tasks import get_contract_metadata_task
 
 
 class EventsService:
 
+    def get_contracts_from_data(self, data: HexStr | None) -> set[ChecksumAddress]:
+        """
```

Comment:
The goal of this function is decode data to get something, in this case the contract addresses. 
For me makes more sense on data-decoder, even as Class method or static method if don´t need anything from the class.

Reply:
Agree that's decoding, but in my opinion it should go here. Decoder is already quite full of logic, and this function can be extended in the future to decode more things than MultiSend. This function is only useful for the events service and not decoding data, parameters, etc.